account/signer.go

```diff
@@ -13,13 +14,23 @@ type Signer struct {
 	publicKey string
 }
 
-func NewSigner(privateKey *big.Int) (*Signer, error) {
+func NewSigner(privateKeyHex string, existingKeystore *MemKeystore) (*Signer, error) {
+	privateKey, ok := new(big.Int).SetString(privateKeyHex, 16)
```

Comment:
Instead of "privateKeyHex" let's go back to the "privateKey" param name, because if we pass 0 to the "SetString" function as the base parameter instead of 16 the function dynamically assumes the current base based on the string prefix. No need to be only a hex value. Ref: https://pkg.go.dev/math/big#Int.SetString

Reply:
done